tools/rlimit/rlimit.c

```diff
@@ -202,30 +216,34 @@ main (int argc, char **argv)
       /* ignore SIGTTOU so that tcsetpgrp call does not block. */
       signal (SIGTTOU, SIG_IGN);
 
-      /* Set child process as foreground process group so that
-       * children can read from the terminal if needed. Note that
-       * we ignore any error here because stdin might not be a terminal.
+      /* If run with --foreground process group, rlimit must be able to
+       * read and write from a tty (not only read => STDIN is a tty).
+       *
        * If only stdout is a terminal there are no issues with not
        * being the foreground process as most terminals are configured
        * with TOSTOP off (so SIGTTOU is only emited in case of terminal
        * settings change.
        */
-      tcsetpgrp(0, getpgid(getpid()));
+      if (foreground && tcsetpgrp(0, getpgid(getpid())) == -1) {
+        perror("tcsetpgrp");
+	exit(4);
+      }
 
       /* Restore SIGTTIN and SIGTTOU signal to their original value
        * in order not to impact children processes behaviour. */
       signal (SIGTTIN, SIG_DFL);
-      signal (SIGTTOU, SIG_DFL);
-      execvp ((const char *) argv[2], (char *const *) &argv[2]);
-      fprintf (stderr, "rlimit: could not run \"%s\": ", argv[2]);
+      if (foreground)
```

Comment:
add a comment here that when non in foreground we keep SIGTTOU to ignore so that child process can write